docs/src/tutorial/cython_tutorial.rst

```diff
@@ -184,29 +191,29 @@ the array (it won't go out of bounds and give a segmentation fault). ::
 
 If no breaks occurred, it means that we found a prime, and the block of code
 after the ``else`` line 16 will be executed. We add the prime found to ``p``.
-If you find having a else after a for loop strange, just know that it's a
-hidden secret of the python syntax, and actually doesn't exist in C!
-But since Cython is made to be written with the Python syntax, it'll
-work out, as if you wrote Python code, but at C speed in this case.
-If the for...else syntax still confuses you, see this excellent
+If you find having an ``else`` after a ``for-loop`` strange, just know that it's a
+lesser known features of the Python language of the python syntax, and
+actually doesn't exist in C! But since Cython is made to be written with the
+Python syntax, it'll work out, but at C speed in this case.
+If the ``for-else`` syntax still confuses you, see this excellent
 `blog post <https://shahriar.svbtle.com/pythons-else-clause-in-loops>`_. ::
 
     # Let's put the result in a python list:
     result_as_list  = [prime for prime in p[:len_p]]
     return result_as_list
 
-Line 22, before returning the result, we need to convert our C array into a
-Python list, because Python can't read C arrays. Note that Cython handle
-for you the conversion of quite some types between C and Python (you can
-see exactly which :ref:`here<type-conversion>`. But not C arrays. We can trick
+Line 22, before returning the result, we need to copy our C array into a
+Python list, because Python can't read C arrays. Cython can automatically
+convert many C types from and to Python types, as described in the
+documentation on :ref:`type conversion <type-conversion>`. But not C arrays. We can trick
```

Comment:
> But not C arrays

I actually just remembered that that's not true. Simply returning the array should make Cython copy it into a list. The problem is that returning a sliced array does not work (yet).

I'll make a couple of minor changes here and there.

Reply:
I didn't know Cython could do it. Thanks for the info!